### ISSUE TITLE
How to use pretrained models for prediction? Is there a tutorial?

### ISSUE THREAD

**Issue by new2scala** [Franck-Dernoncourt/NeuroNER#13]
Have been trying for 2 days with little progress. Seems that the tool works only with this specific format:

```
Japan NNP B-NP B-LOC
began VBD B-VP O
the DT B-NP O
defence NN I-NP O
of IN B-PP O
their PRP$ B-NP O
Asian JJ I-NP B-MISC
Cup NNP I-NP I-MISC
...

```
Ideally, the input for the prediction mode should be plain text without format requirement. I've tried to use **spacy** to convert plain text to the format. This is what I got so far (no idea how to generate the 3rd column):

```
Google NNP B-PROPN O
’s NNP B-PROPN O
second JJ B-ADJ O
generation NN B-NOUN O
TPU NNP B-PROPN O
chips NNS B-NOUN O
...
```

And it seems that the result actually depends on the supplied label (the 4th column, contrary to what's said the documentation), but I'm probably missing something here...

**Comment by Franck-Dernoncourt** [comment]
> the input for the prediction mode should be plain text without format requirement.

Actually, this corresponds to the BRAT format with no annotation. 

>  the result actually depends on the supplied label

You should set in the [`src/parameters.ini`](https://github.com/Franck-Dernoncourt/NeuroNER/blob/master/src/parameters.ini) configuration file:

```
train_model = False
use_pretrained_model = True
```

-------

Here is some instructions on how to use the `trained_models/conll_2003_en` pretrained model on some new unannotated text:

1. Create a file `[NeuroNER folder]/data/dataset/deploy/phrase.txt` that contains `I have been trying for 2 days with little progress in the United States. Seems that the tool works only with this specific format.`
2. Change the following in `[NeuroNER folder]/src/parameters.ini`: 
```[mode]
# At least one of use_pretrained_model and train_model must be set to True.
train_model = False
use_pretrained_model = True
pretrained_model_folder = ../trained_models/conll_2003_en

[dataset]
dataset_text_folder = ../data/dataset
```
3. Run `python3.5 main.py` (it should take one or two minutes)
4. In the output folder (e.g., `[NeuroNER folder]/output/dataset_2017-05-20_14-54-04-457488`), you should find a file `000_deploy.txt` containing:
```
I phrase 0 1 O O
have phrase 2 6 O O
been phrase 7 11 O O
trying phrase 12 18 O O
for phrase 19 22 O O
2 phrase 23 24 O O
days phrase 25 29 O O
with phrase 30 34 O O
little phrase 35 41 O O
progress phrase 42 50 O O
in phrase 51 53 O O
the phrase 54 57 O O
United phrase 58 64 O B-LOC
States phrase 65 71 O I-LOC
. phrase 71 72 O O

Seems phrase 73 78 O O
that phrase 79 83 O O
the phrase 84 87 O O
tool phrase 88 92 O O
works phrase 93 98 O O
only phrase 99 103 O O
with phrase 104 108 O O
this phrase 109 113 O O
specific phrase 114 122 O O
format phrase 123 129 O O
. phrase 129 130 O O
```

Please let you know if that answers your question!

Point taken, we should make it clearer in the documentation.

**Comment by new2scala** [comment]
Yes. Now it works!

```
IBM phrase 0 3 O B-ORG
and phrase 4 7 O O
Google phrase 8 14 O B-ORG
’s phrase 14 16 O O
second phrase 17 23 O O
generation phrase 24 34 O O
TPU phrase 35 38 O B-ORG
chips phrase 39 44 O O
takes phrase 45 50 O O
machine phrase 51 58 O O
learning phrase 59 67 O O
processing phrase 68 78 O O
to phrase 79 81 O O
a phrase 82 83 O O
new phrase 84 87 O O
level phrase 88 93 O O
. phrase 93 94 O O

```

Here's what I missed:
1. Instead of create a text file in the `deploy` folder, I created a file named `deploy.txt`;
2. don't know where the prediction result is stored (going through the `[NeuroNER folder]/output/` folder, I actually found the results from my previous runs last night)

Many thanks, Franck - you just saved my day!

**Comment by Franck-Dernoncourt** [comment]
Thanks for the feedback!

**Comment by spate141** [comment]
@Franck-Dernoncourt How can I load a pre-train model in memory, and classify new sentences on the go? I have followed this approach, and I can classify sentences if I have them in proper folder/file location. 

I was wondering if I can keep the models loaded in memory and somehow can call `def predict(self, text)` in neuroner.py file to classify a list of sentences?

**Comment by trinh-hoang-hiep** [comment]
Did you make changes to predict method to accept list of text and return back all results without saving them to temp directory?. Let me know how you did. Thanks! @new2scala
